sentry/src/main/java/io/sentry/SpanContext.java

```diff
@@ -0,0 +1,126 @@
+package io.sentry;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.protocol.SentryId;
+import io.sentry.util.CollectionUtils;
+import io.sentry.util.Objects;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@Open
+public class SpanContext implements Cloneable {
+  public static final String TYPE = "trace";
+
+  /** Determines which trace the Span belongs to. */
+  private final @NotNull SentryId traceId;
+
+  /** Span id. */
+  private final @NotNull SpanId spanId;
+
+  /** Id of a parent span. */
+  private final @Nullable SpanId parentSpanId;
+
+  /** If trace is sampled. */
+  private transient @Nullable Boolean sampled;
+
+  /** Short code identifying the type of operation the span is measuring. */
+  protected @Nullable String op;
+
+  /**
+   * Longer description of the span's operation, which uniquely identifies the span but is
+   * consistent across instances of the span.
+   */
+  protected @Nullable String description;
+
+  /** Describes the status of the Transaction. */
+  protected @Nullable SpanStatus status;
+
+  /** A map or list of tags for this event. Each tag must be less than 200 characters. */
+  protected @NotNull Map<String, String> tags = new ConcurrentHashMap<>();
```

Comment:
This being concurrent makes me wonder to what degree this code was implemented with concurrency in mind. To what I understood so far no of this stuff was designed to be accessed concurrently, right? The tags here only will? Could you please elaborate?

Reply:
It was designed with concurrency in mind or at least I wanted it to be concurrent. If I missed certain places we can address them individually.